Userland/Utilities/cp.cpp

```diff
@@ -72,10 +74,63 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     bool destination_is_existing_dir = FileSystem::is_directory(destination);
 
     for (auto& source : sources) {
+        // FIXME: May be formatted wrong if destination is a directory with a trailing slash, e.g. if called as such:
+        // cp source dest/
+        // ... `destination_path` may be formatted as `dest//source`, which isn't strictly correct.
         auto destination_path = destination_is_existing_dir
             ? ByteString::formatted("{}/{}", destination, LexicalPath::basename(source))
             : destination;
```

Comment:
```suggestion
       auto destination_path = destination_is_existing_dir
            ? ByteString::formatted("{}/{}", destination.trim("/"sv, TrimMode::Right), LexicalPath::basename(source))
            : destination;
```
Would doing something like this fix the issue?

Reply:
That could work, or possibly something like this:
```cpp
auto destination_path = destination_is_existing_dir
    ? LexicalPath::canonicalized_path(ByteString::formatted("{}/{}", destination, LexicalPath::basename(source)))
    : destination;
 ```

Reply:
That would be better, yes.

Reply:
Would it possibly be a good idea to canonicalise the path in other places as well? E.g. on line 94:
```cpp
if (verbose)
    outln("'{}' -> '{}'", source, destination_path);
```

Would it be a good idea to canonicalise the path here? It would help if the user gave 'dest///subdest///subsub/' as a path, but maybe if they input that it shouldn't be changed?

Reply:
LexicalPath should be able to do any canonicalization you need, unless I'm misunderstanding the question

Reply:
I was wondering whether input path should be canonicalised at all, since it might be confusing to change what they input?